extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridStringArrayConverter.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to string[] parameter.
+    /// </summary>
+    [SupportedConverterType(typeof(string[]))]
+    internal class EventGridStringArrayConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.TargetType != typeof(string[]))
+            {
+                return new(ConversionResult.Unhandled());
+            }
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource is null)
+                {
+                    return new(ConversionResult.Unhandled());
+                }
+
+                var jsonData = JsonSerializer.Deserialize(contextSource, typeof(List<object>)) as List<object>;
```

Comment:
Why not just Deserialize directly to `List<string>` using the generic Deserialize overload?

Reply:
I get a JSON exception if I do `JsonSerializer.Deserialize(contextSource, typeof(string[]))`. Is that what you meant?

Reply:
Ah ok - we are dealing with JSON objects so that won't work. We can still use an object array instead in case System.Text.Json is able to optimize that and we avoid the subsequent ToArray call on L47.